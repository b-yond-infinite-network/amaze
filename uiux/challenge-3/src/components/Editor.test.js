import Editor from './Editor';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router'
import configureMockStore from 'redux-mock-store'
import middlewares from './../store'

const initialState = {
    editor: {
    }
};

describe('Editor Component', () => {

    let mockStore;

    let tree;
    let wrapper;
    let render;

    beforeAll(() => {

        mockStore = configureMockStore(middlewares)(initialState);

        const props = {
            match: {
                params: {
                    slug: ''
                }
            }
        };

        tree = <Provider store={mockStore}>
            <MemoryRouter initialEntries={["/"]}>
                <Editor {...props} />
            </MemoryRouter>
        </Provider>;
        wrapper = mount(tree);
        console.log(wrapper.debug());
        render = renderer.create(tree).toJSON();

    });

    it('Should match the snapshot', () => {
        expect(render).toMatchSnapshot();
    });

    it('Should render without errors', () => {
        const component = wrapper.find('#editor-page');
        expect(component.length).toBe(1);
    });

});