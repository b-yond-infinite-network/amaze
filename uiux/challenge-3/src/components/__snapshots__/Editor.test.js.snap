// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor Component Should match the snapshot 1`] = `
<div
  className="editor-page"
  id="editor-page"
>
  <div
    className="container page"
  >
    <div
      className="row"
    >
      <div
        className="col-md-10 offset-md-1 col-xs-12"
      >
        <form>
          <fieldset>
            <fieldset
              className="form-group"
            >
              <input
                className="form-control form-control-lg"
                onChange={[Function]}
                placeholder="Article Title"
                type="text"
              />
            </fieldset>
            <fieldset
              className="form-group"
            >
              <input
                className="form-control"
                onChange={[Function]}
                placeholder="What's this article about?"
                type="text"
              />
            </fieldset>
            <fieldset
              className="form-group"
            >
              <textarea
                className="form-control"
                onChange={[Function]}
                placeholder="Write your article (in markdown)"
                rows="8"
              />
            </fieldset>
            <fieldset
              className="form-group"
            >
              <input
                className="form-control"
                onChange={[Function]}
                onKeyUp={[Function]}
                placeholder="Enter tags"
                type="text"
              />
              <div
                className="tag-list"
              />
            </fieldset>
            <button
              className="btn btn-lg pull-xs-right btn-primary"
              onClick={[Function]}
              type="button"
            >
              Publish Article
            </button>
          </fieldset>
        </form>
      </div>
    </div>
  </div>
</div>
`;
